### PR TITLE
Fix panic on "unreachable code" that is reachable

### PR DESCRIPTION
**Pull request by dralley**
Fixes https://github.com/servo/servo/issues/22580

### CODE REVIEW

**Comment by dralley** [comment]
According to the discussion on the linked issue, cancellation became supported in this issue https://github.com/servo/servo/issues/22367

This is my first servo PR :)

File: audio/param.rs [hunk 1]
```diff
@@ -327,9 +327,10 @@ impl AutomationEvent {
             AutomationEvent::SetValueCurveAtTime(_, start, _) => start,
             AutomationEvent::RampToValueAtTime(_, _, tick) => tick,
             AutomationEvent::SetTargetAtTime(_, start, _) => start,
-            AutomationEvent::CancelAndHoldAtTime(t) => t,
-            AutomationEvent::CancelScheduledValues(..) | AutomationEvent::SetValue(..) => {
-                unreachable!("CancelScheduledValues/SetValue should never appear in the timeline")
+            AutomationEvent::CancelAndHoldAtTime(tick) => tick,
+            AutomationEvent::CancelScheduledValues(tick) => tick,
+            AutomationEvent::SetValue(..) => {
+                unimplemented!("SetValue should never appear in the timeline")
```
Comment (dralley) [inline audio/param.rs]: One thing I was not sure of is whether ```SetValue``` is something that is *actually* supposed to be unreachable or if, like ```CancelScheduledValues```, it was just not implemented yet.

File: audio/param.rs [hunk 2]
```diff
@@ -327,9 +327,10 @@ impl AutomationEvent {
             AutomationEvent::SetValueCurveAtTime(_, start, _) => start,
             AutomationEvent::RampToValueAtTime(_, _, tick) => tick,
             AutomationEvent::SetTargetAtTime(_, start, _) => start,
-            AutomationEvent::CancelAndHoldAtTime(t) => t,
```
Comment (dralley) [inline audio/param.rs]: This was only changed for naming consistency

**Comment by Manishearth** [comment]
This isn't the correct fix, these items are not supposed to appear in the timeline, they get dealt with during insertion. The correct fix is to move this return to the outer if https://github.com/dralley/media/blob/d06aac618ed1884eb4a5d0b64a86bd371a60017f/audio/param.rs#L210

**Comment by Manishearth** [comment]
(we haven't implemented the difference between Cancel and CancelHold correctly, but that's for another time.)

**Comment by dralley** [comment]
@Manishearth I'm still getting the issue - is this patch what you had in mind or did I move it to the wrong place?

**Comment by dralley** [comment]
```[dalley@localhost servo]$ ./mach run https://midi.city
%c * Tone.js r7 * 
background: #000; color: #fff
internal error: entered unreachable code: CancelScheduledValues/SetValue should never appear in the timeline (thread AudioRenderThread, at /home/dalley/Devel/servo-dev/media/audio/param.rs:333)
stack backtrace:
```

File: audio/param.rs [hunk 3]
```diff
@@ -206,10 +206,11 @@ impl Param {
                 if self.current_event >= self.events.len() {
                     self.val = self.event_start_value;
                 }
-                // don't actually insert the event
-                return;
             }
+            // don't actually insert the event
+            return;
         }
+        dbg!(":?", &event);
```
Comment (dralley) [inline audio/param.rs]: Prints nothing

**Comment by Manishearth** [comment]
@dralley Are you sure it's been recompiled using your local branch? I don't see how this is possible at all.

**Comment by dralley** [comment]
@Manishearth yup I'm sure

```
[dalley@localhost servo]$ ./mach build --dev
   Compiling servo-media-audio v0.1.0 (/home/dalley/Devel/servo-dev/media/audio)
   Compiling servo-media v0.1.0 (https://github.com/servo/media#ecd5a7c6)
```

With a debug at the very top, it does print this

```
[dalley@localhost servo]$ ./mach run https://midi.city
[/home/dalley/Devel/servo-dev/media/audio/param.rs:182] ":?" = ":?"
[/home/dalley/Devel/servo-dev/media/audio/param.rs:182] &event = SetValue(
    0.0,
)
%c * Tone.js r7 * 
background: #000; color: #fff
[/home/dalley/Devel/servo-dev/media/audio/param.rs:182] ":?" = ":?"
[/home/dalley/Devel/servo-dev/media/audio/param.rs:182] &event = CancelScheduledValues(
    Tick(
        128,
    ),
)
internal error: entered unreachable code: CancelScheduledValues/SetValue should never appear in the timeline (thread AudioRenderThread, at /home/dalley/Devel/servo-dev/media/audio/param.rs:335)
```

**Comment by Manishearth** [comment]
Can you paste the full backtrace?

**Comment by Manishearth** [comment]
Oh. Right.

https://github.com/servo/media/blob/50ab2271bcb1e9c6c8b099f9c9a9168b0a34e112/audio/param.rs#L333

That _should_ return the inner time. My bad.

**Comment by Manishearth** [comment]
But the others should still panic

**Comment by dralley** [comment]
How about now?

**Comment by Manishearth** [comment]
@bors-servo r+

Feel free to also add a test for this

**Comment by dralley** [comment]
@Manishearth I have no idea how to add a test for this unfortunately, first Servo PR.  I'd be willing to do so but I'd need some guidance

**Comment by Manishearth** [comment]
No biggie. If you want to try, you need to add another example under `examples/` (make sure you add it to `Cargo.toml`. Perhaps copy one of the params examples and make it mess with cancel messages.

**Pull request merged**